tests/test_SklearnNaiveBayesConverter.py

```diff
@@ -1,63 +1,78 @@
+                        allow_failure="StrictVersion(onnxruntime.__version__)"
+                                       "<= StrictVersion('0.2.1')")
 
-    @unittest.skip(reason="Type 'tensor(float)' of input parameter (input) of operator "
-                          "(ConstantOfShape) in node (ConstantOfShape) is invalid.")
+    @unittest.skipIf(StrictVersion(onnx.__version__) <= StrictVersion('1.3'),
+                     'Needs greater opset 9')
     def test_model_bernoulli_nb_binary_classification(self):
-        model, X = self._fit_model_binary_classification(BernoulliNB())
-        model_onnx = convert_sklearn(model, 'bernoulli naive bayes', [('input', FloatTensorType([1, 4]))])
+        model, X = self._fit_model_binary_classification(BernoulliNB(),
+                                                         load_digits())
+        model_onnx = convert_sklearn(model, 'bernoulli naive bayes',
+                                     [('input', FloatTensorType(X.shape))])
         self.assertIsNotNone(model_onnx)
-        dump_data_and_model(X[:5], model, model_onnx, basename="SklearnBinBernoulliNB-OneOff",
-                            allow_failure="StrictVersion(onnxruntime.__version__) <= StrictVersion('0.1.4')")
+        dump_data_and_model(X, model, model_onnx,
+                        basename="SklearnBinBernoulliNB",
+                        allow_failure="StrictVersion(onnxruntime.__version__)"
+                                       "<= StrictVersion('0.2.1')")
 
-    @unittest.skipIf(StrictVersion(onnx.__version__) <= StrictVersion('1.3'), 'Need Greater Opset 9')
     def test_model_multinomial_nb_multiclass(self):
-        model, X = self._fit_model_multiclass_classification(MultinomialNB())
-        model_onnx = convert_sklearn(model, 'multinomial naive bayes', [('input', FloatTensorType([1, 4]))])
+        model, X = self._fit_model_multiclass_classification(MultinomialNB(),
+                                                             load_iris())
+        model_onnx = convert_sklearn(model, 'multinomial naive bayes',
+                                     [('input', FloatTensorType(X.shape))])
         self.assertIsNotNone(model_onnx)
-        dump_data_and_model(X, model, model_onnx, basename="SklearnMclMultinomialNB-OneOff",
-                            allow_failure="StrictVersion(onnxruntime.__version__) <= StrictVersion('0.1.3')")
+        dump_data_and_model(X, model, model_onnx,
+                        basename="SklearnMclMultinomialNB",
+                        allow_failure="StrictVersion(onnxruntime.__version__)"
+                                       "<= StrictVersion('0.2.1')")
 
-    @unittest.skip(reason="Type 'tensor(float)' of input parameter (input) of operator (ConstantOfShape) "
-                          "in node (ConstantOfShape) is invalid.")
+    @unittest.skipIf(StrictVersion(onnx.__version__) <= StrictVersion('1.3'),
+                     'Needs greater opset 9')
```

Comment:
reason statement doesn't align with '(StrictVersion(onnx.__version__) <= StrictVersion('1.3'),'

Reply:
How does it not align? onnx version 1.3 corrsponds to opset 8 (https://github.com/onnx/onnx/blob/master/docs/Versioning.md#released-versions). I am checking here if onnx.__version__ <= 1.3 and if it is, then skip the test with the message 'needs greater opset 9 (that corresponds to onnx.__version__ 1.4.1)'.

Reply:
Sorry, I always was confused by the statement is actually opposite to the conditional statement.